addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -80,6 +80,19 @@ def _get_duration_from_tracking(self, trackings):
         json = defaultdict(lambda: 0)
         previous_date = self.create_date
 
+        # If there is a tracking value to be created, but still in the
+        # precommit values, create a fake one to take it into account.
+        # Otherwise, the duration_tracking value will add time spent on
+        # previous tracked field value to the time spent in the new value
+        # (after writing the stage on the record)
+        if f'mail.tracking.{self._name}' in self.env.cr.precommit.data:
+            if data := self.env.cr.precommit.data.get(f'mail.tracking.{self._name}', {}).get(self.id):
+                if data.get(self._track_duration_field).id != self[self._track_duration_field].id:
```

Comment:
Hello @lase-odoo , just a quick comment rereading this. Are we certain the field is always in data ? as we read the id directly, we should probably a safety check to make sure key is in data in the if :) (thinking of custom overrides and the like...) Cheers!

Reply:
Hello @Odoonan That's a really good point, I am on it ! Thanks a lot for all your suggestions regarding this ticket btw :slightly_smiling_face: